app/src/main/java/com/gst/gusto/repository/ReviewsRepositoryImpl.kt

```diff
@@ -0,0 +1,20 @@
+package com.gst.gusto.repository
+
+import com.gst.gusto.api.ApiResponse
+import com.gst.gusto.datasource.ReviewsDataSource
+import com.gst.gusto.model.InstaReviews
+
+class ReviewsRepositoryImpl(
+    private val reviewsDataSource: ReviewsDataSource
+) : ReviewsRepository {
+    override suspend fun getInstaReview(
+        token: String,
+        reviewId: Long?,
+        size: Int
+    ): ApiResponse<InstaReviews> {
+        return when (val response = reviewsDataSource.getInstaViewReview(token, reviewId, size)) {
+            is ApiResponse.Success -> ApiResponse.Success(response.data.toDomainModel())
+            is ApiResponse.Error -> ApiResponse.Error(response.errorCode, response.errorMessage)
+        }
```

Comment:
ApiResponse 의 data 를 매핑할 수 있는 확장함수를 하나 만들어서 사용하면 좋을 것 같네요 ~
```kotlin
inline fun <R, T> ApiResponse<T>.map(transform: (value: T) -> R): ApiResponse<R> {
    return when (this) {
        is ApiResponse.Success -> ApiResponse.Success(transform(data))
        is ApiResponse.Error -> ApiResponse.Error(response.errorCode, response.errorMessage)
    }
}
```

그럼 이렇게 바꿀 수 있어요.
```suggestion
        return reviewsDataSource.getInstaViewReview(token, reviewId, size).map { it.toDomainModel() }
```